chichiwang/exercise-js
# Webpack Refactor

## Description
Linked to Issue: N/A
Refactor Webpack configuration files. A few configurations broken out for easier composition, a couple files renamed with purpose namespaces. The goal is increased composability and extensibility.

## Changes
* Add [context](https://webpack.js.org/configuration/entry-context/#context) configuration to client and server configs
* Create a `webpack/rules/` directory to house module rules
  * Move sass rules from `webpack/sassRules.js` to `webpack/rules/sass.inline.js` and `webpack/rules/sass.modules.js`
    * Shared loaders housed in `webpack/rules/sass.loaders.js`
  * Move TS file loader rules from `webpack/base.js` to `webpack/rules/typescript.js`
* Rename `webpack/client.js` to `webpack/client.config.js`
* Rename `webpack/server.js` to `webpack/server.config.js`
* Migrate base config rules from `webpack/base.js` to the client and server config files directly
  * Migrate resolve configurations to `webpack/resolve.js`
  * Remove file `webpack/base.js`

## Steps to QA
Ensure Webpack rules/configurations still work as intended by testing the following scripts:
* `npm run dev`
* `npm run prod`
* `npm run watch`
